File: backend/IMIS/IMIS.Domain/IsoStandard.cs
using Base.Primitives;


namespace IMIS.Domain
{
    public class IsoStandard : Entity<long>
    {
        public long? ParentID { get; set; }
        public required int VersionID { get; set; }
        public StandardVersion? Version { get; set; }
        public required string Particulars { get; set; }
        public required string ClauseRef { get; set; }
        public string? Description { get; set; }
        public required bool isActive { get; set; }
    }
}

File: frontend/imis/lib/iso_standard/models/iso_standard.dart
import 'package:imis/standard_version/models/standard_version.dart';
import 'package:json_annotation/json_annotation.dart';

part 'iso_standard.g.dart';

@JsonSerializable()
class IsoStandard {

  final int id;
  final int? parentID;
  final int versionID;
  final StandardVersion? version;
  final String particulars;
  final String clauseRef;
  final String? description;
  final String? rowVersion;
  final bool? isDeleted;
  final bool isActive;
  final List<IsoStandard>? children;

  IsoStandard({
    required this.id,
    this.parentID,
    required this.versionID,
    this.version,
    required this.particulars,
    required this.clauseRef,
    this.description,
    this.rowVersion,
    this.isDeleted = false,
    this.isActive = true,
    this.children,
  }); 

  factory IsoStandard.fromJson(Map<String, dynamic> json) =>
      _$IsoStandardFromJson(json);

  Map<String, dynamic> toJson() => _$IsoStandardToJson(this);
}
